File: code_examples/5-Holding_Data/14-Do_While_Loop/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a do ... while loop
*/

using System;

/*
 * The Program class
*/
class Program {
    static void Main() {
        // this function uses a do ... while loop

        int counter = 0;
        int positiveInteger;

        // input
        Console.Write("Enter how many times to repeat: ");
        positiveInteger = Convert.ToInt32(Console.ReadLine());
        Console.WriteLine();

        // process & output
        do {
            Console.WriteLine(counter + " time(s) through the loop.");
            counter = counter + 1;
        } while (counter < positiveInteger);

        Console.WriteLine("\nDone.");
    }
}

File: code_examples/5-Holding_Data/14-Do_While_Loop/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a do ... while loop

#include <stdio.h>

int main() {
    // this function uses a do ... while loop
    int counter = 0;
    int positiveInteger;

    // input
    printf("Enter how many times to repeat: ");
    scanf("%d", &positiveInteger);
    printf("\n");

    // process & output
    do {
        printf("%d time(s) through the loop.\n", counter);
        counter = counter + 1;
    } while (counter < positiveInteger);

    printf("\nDone.\n");
    return 0;
}
